### PR TITLE
chore(launch): Add Job type, fix typing issues, and have k8s manager use queue driver

### PR DESCRIPTION
**Pull request by KyleGoyette**
Description
-----------

- Fixes WB-NNNNN
- Fixes #NNNN

What does the PR do? Include a concise description of the PR contents.

- [ ] I updated CHANGELOG.md, or it's not applicable

Testing
-------
How was this PR tested?

### CODE REVIEW

File: wandb/sdk/launch/agent2/controllers/k8s.py [hunk 1]
```diff
@@ -64,90 +57,94 @@ def __init__(
...
         run_id = project.run_id
         job_tracker = self.legacy.job_tracker_factory(run_id)
         job_tracker.update_run_info(project)
 
-        assert (
-            project.docker_image is not None
-        ), "Docker image must be set when using k8s"
-        run = await self.legacy.runner.run(project, project.docker_image)
+        ack_result = await self.jobset.ack_job(item.id, run_id)
+        self.logger.info(f"Acked item: {json.dumps(ack_result, indent=2)}")
```
Comment (TimH98) [inline wandb/sdk/launch/agent2/controllers/k8s.py:122]: nit: Move this below the `if not ack_result:` block

File: wandb/sdk/launch/agent2/controllers/k8s.py [hunk 2]
```diff
@@ -64,90 +57,94 @@ def __init__(
...
-        self.logger.info(f"Acked item: {json.dumps(ack_result, indent=2)}")
-        self.active_runs[item["id"]] = run
+        self.active_runs[item.id] = run
         self.logger.info(f"Inside launch_item_task, project.run_id = {run_id}")
         return project.run_id
 
-    async def release_item(self, item: Any) -> Any:
-        del self.active_runs[item["id"]]
-        self.logger.info(f"Releasing item: {json.dumps(item, indent=2)}")
+    async def release_item(self, item: str) -> None:
```
Comment (TimH98) [inline wandb/sdk/launch/agent2/controllers/k8s.py:148]: nit: `item` -> `item_id`

File: wandb/sdk/launch/agent2/controllers/local_container.py [hunk 3]
```diff
@@ -96,36 +98,39 @@ async def reconcile(self):
...
         if not run:
             job_tracker.failed_to_start = True
-            self.logger.error(f"Failed to start run for item {item['id']}")
+            self.logger.error(f"Failed to start run for item {item.id}")
             raise NotImplementedError("TODO: handle this case")
 
-        ack_result = await self.queue_driver.ack_run_queue_item(
-            item["runQueueItemId"], run_id
-        )
-        self.logger.info(f"Acked item: {json.dumps(ack_result, indent=2)}")
```
Comment (TimH98) [inline wandb/sdk/launch/agent2/controllers/local_container.py:126]: Is this meant to be removed or just moved up like in the k8s controller?

File: wandb/sdk/launch/agent2/controllers/k8s.py [hunk 4]
```diff
@@ -64,90 +57,94 @@ def __init__(
...
-        owned_items = [
-            item
+        owned_items = {
+            item.id: item
             for item in raw_items
-            if item["state"] in ["LEASED", "CLAIMED", "RUNNING"]
-            and item["launchAgentId"] == self.config["agent_id"]
-        ]
-        pending_items = [item for item in raw_items if item["state"] in ["PENDING"]]
+            if item.state in ["LEASED", "CLAIMED"]
```
[the remark this answers is not in the document]
Comment (KyleGoyette) [reply wandb/sdk/launch/agent2/controllers/k8s.py:75]: AFAIK there is no Running state for run queue items, unless we added that in the JobSetDiff?

File: wandb/sdk/launch/agent2/controllers/k8s.py (hunk 1 above)
Comment (TimH98) [reply wandb/sdk/launch/agent2/controllers/k8s.py:122]: (The GitHub UI doesn't make it obvious but I'm only referring to the `self.logger.info` line here)

Comment (KyleGoyette) [reply wandb/sdk/launch/agent2/controllers/k8s.py:122]: nice catch

File: wandb/sdk/launch/agent2/controllers/local_container.py (hunk 3 above)
Comment (KyleGoyette) [reply wandb/sdk/launch/agent2/controllers/local_container.py:126]: Oof, I swear that I moved it...

**Review by TimH98 — APPROVED** [review]
sweet

**Pull request merged**